src/search.c

```diff
@@ -45,112 +46,113 @@
 #include "search.h"
 #include "checksum.h"
 
-#define TABLE_SIZE (1<<16)
-#define NULL_TAG (-1)
+// maximum load factor of hash table
+#define LOAD_FACTOR_NUM 7
+#define LOAD_FACTOR_DEN 10
 
-#define gettag2(s1,s2) (((s1) + (s2)) & 0xFFFF)
-#define gettag(sum) gettag2((sum)&0xFFFF,(sum)>>16)
-
-static void swap(rs_target_t *t1, rs_target_t *t2) {
-    unsigned short ts = t1->t;
-    t1->t = t2->t;
-    t2->t = ts;
-
-    int ti = t1->i;
-    t1->i = t2->i;
-    t2->i = ti;
+static unsigned int next_pow_2(unsigned int v)
+{
+    if (v < 2) return v;
+    return (1U << (32 - __builtin_clz(v - 1)));
 }
 
-static int rs_compare_targets(rs_target_t const *t1, rs_target_t const *t2, rs_signature_t * sums) {
-    int v = (int) t1->t - (int) t2->t;
-    if (v != 0)
-        return v;
-
-    rs_weak_sum_t w1 = sums->block_sigs[t1->i].weak_sum;
-    rs_weak_sum_t w2 = sums->block_sigs[t2->i].weak_sum;
-
-    v = (w1 > w2) - (w1 < w2);
-    if (v != 0)
-        return v;
-
-    return memcmp(sums->block_sigs[t1->i].strong_sum,
-            sums->block_sigs[t2->i].strong_sum,
-            sums->strong_sum_len);
+// mix full 32-bits of bad adler32 hash before truncating
+static unsigned int mix32(unsigned int h)
+{
+    // final mix function of MurmurHash2
```

Comment:
It'd be good to be more clear here what relation this has to the adler hash, if any - I think nothing, this is a separate hash just for the table?